functions/src/index.test.ts

```diff
@@ -1,5 +1,45 @@
-describe("Hello World", () => {
-  test("Placeholder Test", () => {
-    expect("Hello").toEqual("Hello");
+import "jest";
+import fetch from "node-fetch";
+import { API_BASE_URL, createUniqueId } from "./utils";
+import { getShareLinkErrorByCode, ShareLinkErrorCode } from "./error-handling";
+
+const TEST_PAYLOAD = {
+  url: "https://www.covidactnow.org",
+  imageUrl: "https://covidactnow-prod.web.app/share/4047-2795/home.png",
+  title: "Covid Act Now - America's Covid Tracking Dashboard.",
+  description: "See how Covid is spreading in your community.",
+};
+
+beforeAll(() => {
+  if (!process.env.FUNCTIONS_EMULATOR) {
+    throw new Error("Test suite must be run with the Firebase emulator.");
+  }
+});
+
+describe("POST /registerUrl/", () => {
+  test("it returns a 200 and creates a share link", async () => {
+    const response = await fetch(`${API_BASE_URL}/registerUrl`, {
+      method: "POST",
+      body: JSON.stringify(TEST_PAYLOAD),
+      headers: { "Content-Type": "application/json" },
+    });
```

Comment:
FWIW- If we find ourselves duplicating code across tests I could imagine some helpers for creating these requests (with the right method / headers, etc.).